app/assets/stylesheets/views/_homepage.scss

```diff
@@ -216,108 +216,134 @@ body.homepage {
-      padding-bottom: 30px;
+      @extend %contain-floats;
+      margin-bottom: 30px;
 
-      .header-search-content {
-        @extend %contain-floats;
-        position: relative;
-        background: #fff;
-      }
       label {
-        float: left;
-        height: 40px;
-        line-height: 40px;
-        text-indent: 15px;
-        overflow: hidden;
-        display: block;
-        margin-right: 10px;
-        color: $secondary-text-colour;
+        color: $white;
+      }
 
-        .js-enabled & {
-          float: none;
-          position: absolute;
-          left: 0;
-          top: 1px;
-          z-index: 1;
+      .search-input-wrapper {
+        float: left;
+        width: 100%;
+        
+        .search-input {
           width: 100%;
-          margin-right: 0;
+          margin: 0.5em 0;
+          padding: 6px;
+
+          @include core-19($line-height: (28 / 19), $line-height-640: (28 / 16));
         }
       }
-      input {
-        @include box-sizing(border-box);
+
+      .submit {
         float: left;
-        width: 50%;
-        min-width: 0;
-        display: block;
-        margin: 0;
+        padding: 6px;
+        background: $black;
+        color: $white;
         border: 0;
-        height: 40px;
-        padding: 6px 0;
-        @include ie-lte(7){
-          height: 28px;
-        }
-        @include appearance(none);
       }
 
-      input#search-main {
-        @include core-19($line-height: (28/19), $line-height-640: (28/16));
-        position: relative;
-        padding: 6px;
-        z-index: 3;
-        background: transparent;
+      // IE6 + IE7 always get the simpler version, regardless of whether
+      // javascript is enabled
+      @if ($is-ie == false) or ($ie-version >= 8) {
+        .search-input-wrapper {
+          @include box-sizing(border-box);
 
-        &.focus,
-        &:focus {
-          background: #fff;
-        }
+          padding-right: 40px;
+          margin-right: -40px;
 
-        @include ie-lte(7){
-          padding-left: 0;
-        }
-        .js-enabled & {
-          width: 86%;
-          @include calc(width, "100% - 37px");
-        }
-      }
+          .search-input {
+            @include box-sizing(border-box);
+            @include appearance(none);
 
-      input.submit {
-        position: absolute;
-        z-index: 4;
-        right: 0;
-        top: 0;
-        width: 40px;
-        height: 40px;
+            position: relative;
+            height: 40px;
+            margin: 0;
 
-        border: 1px solid $black;
-        border-width: 0 0 0 1px;
-        @include ie-lte(7){
-          border-width: 1px;
+            border: 0;
+            outline-offset: 0;
+          }
         }
-        border-left-color: #222;
-        @include border-radius(0);
 
-        overflow: hidden;
-        text-indent: -5000px;
-        cursor: pointer;
+        .submit {
+          @include box-sizing(border-box);
+          float: left;
```

Comment:
Minor: It is already set to `float: left` above, so would inherit it and doesn't need the line. I'm not sure it's necessary to float at all? (Haven't tested in all the browsers, so some might need it?)